### ISSUE TITLE
conda-build fails with encoding "None"

### ISSUE THREAD

**Issue by pitrou** [conda/conda-build#636]
This has just happened on our llvmlite Jenkins builders and seems to be a regression:

```
+ conda build -q -c numba conda-recipes/llvmlite-jenkins --no-binstar-upload
Removing old build environment
Removing old work directory
BUILD START: llvmlite--py26_0
Fetching package metadata: ............
Solving package specifications: ....................
The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    system-5.8                 |                2         170 KB
    llvmdev-3.6.2              |                1        61.4 MB
    python-2.6.9               |                1        10.7 MB
    ordereddict-1.1            |           py26_0           4 KB
    setuptools-18.0.1          |           py26_0         343 KB
    enum34-1.0.4               |           py26_0          48 KB
    pip-7.1.0                  |           py26_0         1.4 MB
    ------------------------------------------------------------
                                           Total:        74.0 MB

The following NEW packages will be INSTALLED:

    enum34:      1.0.4-py26_0 
    llvmdev:     3.6.2-1      
    openssl:     1.0.1k-1     
    ordereddict: 1.1-py26_0   
    pip:         7.1.0-py26_0 
    python:      2.6.9-1      
    readline:    6.2-2        
    setuptools:  18.0.1-py26_Cloning into bare repository '/var/jenkins/workspace/conda_llvmlite/CONDA_PY/26/label/CentOS5x64b/miniconda/conda-bld/git_cache/__github.com_numba_llvmlite.git'...
Cloning into '/var/jenkins/workspace/conda_llvmlite/CONDA_PY/26/label/CentOS5x64b/miniconda/conda-bld/work'...
done.
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda-build/issues

Include the output of the command 'conda info' in your report.


0
    sqlite:      3.8.4.1-1    
    system:      5.8-2        
    tk:          8.5.18-0     
    zlib:        1.2.8-0      

Removing old work directory
Traceback (most recent call last):
  File "/var/jenkins/workspace/conda_llvmlite/CONDA_PY/26/label/CentOS5x64b/miniconda/bin/conda-build", line 5, in <module>
    sys.exit(main())
  File "/var/jenkins/workspace/conda_llvmlite/CONDA_PY/26/label/CentOS5x64b/miniconda/lib/python2.7/site-packages/conda_build/main_build.py", line 190, in main
    args_func(args, p)
  File "/var/jenkins/workspace/conda_llvmlite/CONDA_PY/26/label/CentOS5x64b/miniconda/lib/python2.7/site-packages/conda_build/main_build.py", line 468, in args_func
    args.func(args, p)
  File "/var/jenkins/workspace/conda_llvmlite/CONDA_PY/26/label/CentOS5x64b/miniconda/lib/python2.7/site-packages/conda_build/main_build.py", line 391, in execute
    override_channels=args.override_channels, include_recipe=args.include_recipe)
  File "/var/jenkins/workspace/conda_llvmlite/CONDA_PY/26/label/CentOS5x64b/miniconda/lib/python2.7/site-packages/conda_build/build.py", line 392, in build
    source.provide(m.path, m.get_section('source'))
  File "/var/jenkins/workspace/conda_llvmlite/CONDA_PY/26/label/CentOS5x64b/miniconda/lib/python2.7/site-packages/conda_build/source.py", line 269, in provide
    git_source(meta, recipe_dir)
  File "/var/jenkins/workspace/conda_llvmlite/CONDA_PY/26/label/CentOS5x64b/miniconda/lib/python2.7/site-packages/conda_build/source.py", line 136, in git_source
    git_info()
  File "/var/jenkins/workspace/conda_llvmlite/CONDA_PY/26/label/CentOS5x64b/miniconda/lib/python2.7/site-packages/conda_build/source.py", line 160, in git_info
    stdout = stdout.decode(encoding, 'ignore')
TypeError: decode() argument 1 must be string, not None
Build step 'Execute shell' marked build as failure
Flowdock: Team Inbox notification sent successfully
Notifying upstream projects of job completion
Finished: FAILURE
```

**Comment by msarahan** [comment]
I believe this was fixed by #614 - but that fix came just before the 1.18.0 release.  Is this sufficiently broken to warrant another release?

**Comment by pitrou** [comment]
You mean just after? This breaks our Jenkins machines, so on our side this is a rather annoying issue. Is there a simple workaround?

**Comment by msarahan** [comment]
Yes - edited to change "before" to "after"

As @jakirkham mentioned in #614, one potential workaround is to set the PYTHONIOENCODING environment variable:

PYTHONIOENCODING="UTF-8"

**Comment by pitrou** [comment]
Le 12/10/2015 15:11, Mike Sarahan a écrit :
> As @jakirkham <https://github.com/jakirkham> mentioned in #614
> <https://github.com/conda/conda-build/pull/614>, one potential
> workaround is to set the PYTHONIOENCODING environment variable:
> 
> PYTHONIOENCODING="UTF-8"

This seems to work, thank you!

**Comment by rmcgibbo** [comment]
yeah, we're hitting this too. a bugfix release would be awesome, if it's not too much trouble (also looking forward to numpy 1.10).

**Comment by jakirkham** [comment]
Glad to hear the workaround is helping people.

>  Is this sufficiently broken to warrant another release?

At some point a bug fix release would be nice, but the workaround isn't too painful so that is what I am using at present. I will leave the release timing up to your discretion.

**Comment by msarahan** [comment]
We are preparing for the 2.4 release of Anaconda at the end of this month.
We will work this bugfix release in as soon as possible, but at the very
latest, for the 2.4 release of Anaconda.

**Comment by mcg1969** [comment]
I think I just encountered this bug on trying to use Anaconda.org's build workers...

**Comment by msarahan** [comment]
The fixes in #614 were released in Conda-build 1.18.1.  Please file a new issue if this problem recurs.
